lib/bolt/pal.rb

```diff
@@ -157,7 +157,7 @@ def in_bolt_compiler
         Puppet.override(bolt_project: bolt_project,
                         bolt_project_data: bolt_project_data,
                         yaml_plan_instantiator: Bolt::PAL::YamlPlan::Loader) do
-          pal.with_script_compiler do |compiler|
+          pal.with_script_compiler(disable_facts: true) do |compiler|
```

Comment:
@puppetlabs/skeletor I think this will also need to update here: https://github.com/puppetlabs/orchestrator/blob/70e55aae51084cf08c32515a0cd98bb4eb6ad86c/ruby/bolt-plan-executor-lib/bolt_plan_executor/pe_pal.rb#L119 For context, we had to revert removing $facts / trusted facts / server facts from the script compiler in Puppet PAL because it broke Choria. For now at least we're just going to add a `disable_facts` keyword argument to `with_script_compiler` to tell Puppet to not load those in the Bolt context (see https://github.com/puppetlabs/puppet/pull/8268).

Reply:
Updated this to be `set_local_facts` instead of `disable_facts` per @nicklewis 's review: https://github.com/puppetlabs/puppet/pull/8268#discussion_r469429071